src/Views/DogDetail.js

```diff
@@ -0,0 +1,40 @@
+// import { Link } from 'react-router-dom';
+
+// export default function DogDetail() {
+//   return <h2> Test Dog Detail</h2>;
+// }
+import DogInfo from '../component/Dog';
+import { fetchDogs } from '../services/DogRoute';
+import { useEffect, useState } from 'react';
+import { getDogsById } from '../services/DogRoute';
+import { useParams } from 'react-router-dom';
+
+export default function Home() {
+  const { id } = useParams();
+  const [info, setInfo] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
```

Comment:
There are two useEffects trying to do almost the same thing and introducing a bug where the dog details won't always show. Removing the first useEffect and setting loading to false in the second useEffect should clear that up.

Reply:
Removed useEffect and moved loading to new useEffect